### PR TITLE
Serializing srvtopo aggregate updates.

### PR DESCRIPTION
**Pull request by alainjobart**
Using a channel to serialize updates. Otherwise, when two tablets update
their records at the same time, the resulting aggregate stats may appear
out of order, and not be correct. This introduced flakiness in our tests
(detected internally by our build tools, at least, haven't checked
travis).

### CODE REVIEW

File: go/vt/discovery/tablet_stats_cache.go [hunk 1]
```diff
@@ -357,21 +361,29 @@ func (tc *TabletStatsCache) updateAggregateMap(keyspace, shard string, tabletTyp
 	newAgg := MakeAggregateMap(stats)
 	e.aggregates = newAgg
 
-	// And broadcast the change in the background.
-	go func() {
+	// And broadcast the change in the background, if we need to.
+	if !tc.tsm.HasSubscribers() {
```
Comment (michael-berlin) [inline go/vt/discovery/tablet_stats_cache.go]: This is missing the `RLock` on `tc.mu`?

File: go/vt/discovery/tablet_stats_cache.go [hunk 2]
```diff
@@ -357,21 +361,29 @@ func (tc *TabletStatsCache) updateAggregateMap(keyspace, shard string, tabletTyp
...
+		ter = stats[0].TabletExternallyReparentedTimestamp
+	}
+	diffs := MakeAggregateMapDiff(keyspace, shard, tabletType, ter, oldAgg, newAgg)
+	tc.aggregatesChan <- diffs
+}
+
+// broadcastAggregateStats is called in the background to send aggregate stats
+// in the right order to our subscribers.
+func (tc *TabletStatsCache) broadcastAggregateStats() {
+	for diffs := range tc.aggregatesChan {
```
Comment (michael-berlin) [inline go/vt/discovery/tablet_stats_cache.go]: Is it okay that you're never going to close this channel?

Will the binaries shut down cleanly anyway?

File: go/vt/discovery/tablet_stats_cache.go (hunk 1 above)
Comment (alainjobart) [reply go/vt/discovery/tablet_stats_cache.go]: Good catch!

File: go/vt/discovery/tablet_stats_cache.go (hunk 2 above)
Comment (alainjobart) [reply go/vt/discovery/tablet_stats_cache.go]: They do, yes. There is no shutdown at this point for this code, and we have other channels in other places that are never closed.

**Pull request merged**